flocker/node/test/test_deploy.py

```diff
@@ -2196,6 +2196,38 @@ def test_unknown_applications(self):
         expected = sequentially(changes=[])
         self.assertEqual(expected, changes)
 
+    def test_different_node_is_ignorant(self):
+        """
+        The fact that a different node is ignorant about its manifestations
+        does not prevent calculating changes necessary for the current
+        node.
+        """
+        node_state = NodeState(
+            hostname=u"10.1.1.1",
+            uuid=uuid4(),
+            manifestations={MANIFESTATION.dataset_id:
+                            MANIFESTATION},
+            devices={}, paths={},
```

Comment:
Not necessarily. ZFS doesn't have devices. For now this is fine, I think, improving manifests will happen some other time.